File: src/Explorer.Core/Models/CommandDefinition.cs
namespace NoctusExplorer.Core.Models;

public sealed class CommandDefinition
{
    public required string Id { get; init; }
    public required string Name { get; init; }
    public string Description { get; init; } = "";
    public KeyChord? DefaultBinding { get; init; }
    public string? IconId { get; init; }
    public required Func<bool> CanExecute { get; init; }
    public required Action Execute { get; init; }
}

File: macos/Sources/NoctusCore/Models/CommandDefinition.swift
import Foundation

public final class CommandDefinition {
    public let id: String
    public let name: String
    public let description: String
    public let defaultBinding: KeyChord?
    public let canExecute: () -> Bool
    public let execute: () -> Void

    public init(id: String, name: String, description: String = "",
                defaultBinding: KeyChord? = nil,
                canExecute: @escaping () -> Bool = { true },
                execute: @escaping () -> Void) {
        self.id = id
        self.name = name
        self.description = description
        self.defaultBinding = defaultBinding
        self.canExecute = canExecute
        self.execute = execute
    }
}
